goupnp.go

```diff
@@ -125,15 +125,25 @@ func DeviceByURL(ctx context.Context, loc *url.URL) (*RootDevice, error) {
 // but should not be changed after requesting clients.
 var CharsetReaderDefault func(charset string, input io.Reader) (io.Reader, error)
 
-// HTTPClient is the client used inside of requestXml for fetching XML from a specific endpoint.
-var HTTPClient = http.DefaultClient
+// contextKey is an unexported type which prevents construction of other contextKeys.
+type httpContextKey struct{}
+
+// WithHTTPClient returns a context wrapping ctx with the HTTP client set to c.
+// c must be non-nil.
```

Comment:
With your implementation below, this would be:

// If c is nil, http.DefaultClient is used.

Reply:
These docs still don't match the implementation. Please make them consistent one way or another.